### DISCUSSION TITLE
Ngalert Math expression need support more syntax

### DISCUSSION THREAD

**Discussion by homerzhou** [grafana/grafana#48845] [Alerting]
Scenario:

1. Two host, host1 and host2
2. prometheus use node_exporter collect cpu metrics
3. get metrics example
    cpu{host="host1"} 10
    cpu{host="host2"} 80
4. Use grafana nglert configure rule to create alert, host1's cpu set threshold with 20, host2's cpu set threshold with 90

Current solution:

I have to create two alert rule, blow
1. For host1's cpu set threshold with 20
A: Prometheus query:   cpu{host="host1"}
B: operation: Reduce,   Fuction: Max,   Input: A
C: operation: Math,  Expression: $B > 20

2. For host2's cpu set threshold with 90
A: Prometheus query:   cpu{host="host2"}
B: operation: Reduce,   Fuction: Max,   Input: A
C: operation: Math,  Expression: $B > 90

Question:

If I Have the so many host machine, the ngalert rule will be too much for each host

Expection:

Operation Math's Expression should support more syntax
Example blow:
A: Prometheus query:  cpu
B: operation: Reduce,   Fuction: Max,   Input: A
C: operation: MathMatch,  Expression:  if ($metric.label.host=="host1") return $metric.value > 20 elseif ($metric.label.host=="host2") return $metric.value > 90 else return $metric.value > 50

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
